sklego/meta/regression_outlier_detector.py

```diff
@@ -112,7 +129,8 @@ def fit(self, X, y=None):
         ValueError
             If the `model` is not a regression estimator.
         """
-        self.idx_ = np.argmax([i == self.column for i in X.columns]) if isinstance(X, pd.DataFrame) else self.column
+        X = nw.from_native(X, eager_only=True, strict=False)
+        self.idx_ = np.argmax([i == self.column for i in X.columns]) if isinstance(X, nw.DataFrame) else self.column
         X = check_array(X, estimator=self)
```

Comment:
How does `check_array` behave on narwhals frame? Converts it to numpy array?

Reply:
That's a good point actually, thanks for having asked! I just digged into it, but admittedly I should have checked it more carefully in the first place

Yes, just like for pandas/polars input directly, it converts to a numpy array. BUT - they have some pandas-specific logic in there. So, we can just pass `nw.to_native(X, strict=False)`, and then we can be sure there'll be no difference with respect to the status quo